apps/condo/domains/notification/adapters/firebaseAdapter.js

```
@@ -0,0 +1,226 @@
+const admin = require('firebase-admin')
+const isEmpty = require('lodash/isEmpty')
+const isNull = require('lodash/isNull')
+const faker = require('faker')
+
+const conf = require('@core/config')
+
+const { PUSH_FAKE_TOKEN_SUCCESS, PUSH_FAKE_TOKEN_FAIL } = require('../constants/constants')
+
+const FIREBASE_CONFIG = conf['FIREBASE_CONFIG_JSON'] && JSON.parse(conf['FIREBASE_CONFIG_JSON'])
+const EMPTY_RESULT = {
+    responses: [],
+    successCount: 0,
+    failureCount: 0,
+}
+const FAKE_ERROR_RESPONSE = {
+    success: false,
+    error: {
+        errorInfo: {
+            code: 'fake-error',
+            message: 'Fake error mesage',
+        },
+    },
+}
+
+const getFakeSuccessResponse = () => ({
+    success: true,
+    messageId: `fake-success-message/${faker.datatype.uuid()}`,
+})
+
+/**
+ * Send push notification to pushToken via app, configured by FIREBASE_CONFIG in .env
+ * Attention! Notifications could only be sent to devices, connected via same PROJECT_ID.
+ * Attempts to send push notifications to devices, connected through different projects will fail.
+ */
+class FirebaseAdapter {
+    constructor (config = FIREBASE_CONFIG) {
+        this.app = null
+
+        if (isEmpty(config)) console.error('Valid FIREBASE_CONFIG_JSON should be provided within .env, and can be retrieved from https://console.firebase.google.com/project/__PROJECT_ID__/settings/serviceaccounts/adminsdk')
+
+        try {
+            this.app = admin.initializeApp({ credential: admin.credential.cert(config) })
+        } catch (error) {
+            console.error('Unable to authorize at FireBase using provided FIREBASE_CONFIG_JSON', error)
```

You should throw an error. Don't hide an error